src/button-dropdown/utils/use-button-dropdown.ts

```diff
@@ -31,14 +31,23 @@ export function useButtonDropdown({
   onItemFollow,
   hasExpandableGroups,
   isInRestrictedView = false,
-  usingMouse,
 }: UseButtonDropdownOptions): UseButtonDropdownApi {
-  const { targetItem, isHighlighted, isExpanded, highlightItem, moveHighlight, expandGroup, collapseGroup, reset } =
-    useHighlightedMenu({
-      items,
-      hasExpandableGroups,
-      isInRestrictedView,
-    });
+  const {
+    targetItem,
+    isHighlighted,
+    isFocused,
+    isExpanded,
+    highlightItem,
+    moveHighlight,
+    expandGroup,
+    collapseGroup,
+    reset,
+    setIsUsingMouse,
```

Comment:
If the approach from select/multiselect can be taken, the highlightedType can be then set from inside `highlightItem` so no need in additional callback.

Reply:
In buttondropdown the highlightItem() is called when mouseEnter on item-element, unlike in select it depends on the event on list. In order to change style when mousemove, we need change the event handler and pass usingMosue -- otherwise highlightItem is called before usingMouse gets updated as usingMouse is updated on the dropdown event handler. So it makes more sense to me to not make highlightedType dependent on highlightItem in this case.